### PR TITLE
VDS-2352: Browse list of addresses

### PR DESCRIPTION
**Pull request by ilyawzrd**
Снимок экрана 2021-08-12 в 15 37 38

Description: 

--

QA-test: 

Demo-test: 

Download artifact URL: ghcr.io/virtocommerce/demo-xapi-app:1.0.0-pr-33-vds-2352-7bf4d5f5

### CODE REVIEW

**Review by trueboroda — APPROVED** [review]

File: src/app/graphql/queries/get-organization-addresses.graphql [hunk 1]
```diff
@@ -0,0 +1,27 @@
+query getOrganizationAddresses($id: String!, $count: Int, $cursor: String, $sort: String) {
```
Comment (asvishnyakov) [inline src/app/graphql/queries/get-organization-addresses.graphql:1]: May we use `skip` & `take` names instead?

File: src/app/graphql/queries/get-organization-addresses.graphql [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        countryCode
+        postalCode
+        regionName
+        city
+      }
+      pageInfo {
+        endCursor
+        hasNextPage
+        hasPreviousPage
+        startCursor
```
Comment (asvishnyakov) [inline src/app/graphql/queries/get-organization-addresses.graphql]: Does we really need this information? I don't see this fields are used anywhere in code. Please left only needed fields in this query

File: src/app/modules/addresses/addresses.component.html [hunk 3]
```diff
@@ -0,0 +1,46 @@
+<main class="h-100" role="main">
+  <div class="container pt-4 pb-6">
+    <h1 class="mb-3" *ngIf="curentCustomerOrganization$ | async as curentCustomerOrganization">{{curentCustomerOrganization?.name}}</h1>
```
Comment (asvishnyakov) [inline src/app/modules/addresses/addresses.component.html:3]: You _probably_ need to move this `if` to the container/`div`-wrapper and don't repeat it below

File: src/app/modules/addresses/addresses.component.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+  pageSize = pageSize;
+
+  faChevronLeft = faChevronLeft;
+
+  faChevronRight = faChevronRight;
+
+  sortDescending = 'desc';
+
+  sortAscending = 'asc';
```
Comment (asvishnyakov) [inline src/app/modules/addresses/addresses.component.ts]: May be move it to separate reusable constants file/module?

File: src/app/modules/addresses/addresses.component.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+  addresses?: Address[] | null;
+
+  totalCount?: number | null;
+
+  cursor = '0';
+
+  page = 1;
+
+  pageSize = pageSize;
```
Comment (asvishnyakov) [inline src/app/modules/addresses/addresses.component.ts]: I'll prefer to introduce separate class/inteface for them (and sortDirection), like `PageInfo`, `SearchInfo` or may be `SearchCriteria`

File: src/app/modules/addresses/addresses.component.ts [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+  sortDescending = 'desc';
+
+  sortAscending = 'asc';
+
+  sortDirection = this.sortDescending;
+
+  constructor(private readonly store: Store) { }
+
+  ngOnInit(): void {
+    this.curentCustomerOrganization$.subscribe(value => this.curentCustomerOrganizationId = value?.id);
```
Comment (asvishnyakov) [inline src/app/modules/addresses/addresses.component.ts]: We assume current organization is always exists and we prevent someone go to this page by `if` on dashboard. I'll prefer use `nonNull` type guard here instead (in `pipe`) and remove `if` check below. This component will be broken any way if someone will go here as anonymous user by direct link

File: src/app/modules/addresses/addresses.component.ts [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+      this.loadAddresses(this.curentCustomerOrganizationId, this.pageSize, this.cursor, sortigExpression);
+    }
+
+    this.store.select(selectOrganizationAddresses)
+      .pipe(takeUntil(this.unsubscriber))
+      .subscribe(addresses => {
+        this.addresses = addresses?.items;
+        this.totalCount = addresses?.totalCount;
+      });
+  }
```
Comment (asvishnyakov) [inline src/app/modules/addresses/addresses.component.ts]: May we just use `| async` in html instead?

File: src/app/modules/addresses/addresses.component.ts (hunk 6 above)
Comment (asvishnyakov) [reply src/app/modules/addresses/addresses.component.ts]: Also there is no ubsubscription

File: src/assets/images/arrow-down.svg [hunk 8]
```diff
@@ -0,0 +1,3 @@
+<svg width="12" height="6" viewBox="0 0 12 6" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (asvishnyakov) [inline src/assets/images/arrow-down.svg:1]: Can't we reuse already existing `arrow-*.svg` images?

File: src/styles/_maps.scss [hunk 9]
```diff
@@ -12,7 +12,8 @@ $theme-colors: map-merge(
     'gray-100': $gray-100,
     'gray-modal': $gray-modal-bg,
     'black': $black,
-    'white': $white
+    'white': $white,
+    'table-header': $gray-modal-border
```
Comment (asvishnyakov) [inline src/styles/_maps.scss:16]: Are you sure it should be global theme color?

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: src/assets/images/arrow-down.svg (hunk 8 above)
Comment (ilyawzrd) [reply src/assets/images/arrow-down.svg:1]: @asvishnyakov already existed ones have different color, so there are two ways:
1) We are using new ones and removing the old ones from assets
2) We can use old ones and then write custom css to change the color

File: src/styles/_maps.scss (hunk 9 above)
Comment (ilyawzrd) [reply src/styles/_maps.scss:16]: @asvishnyakov by adding a color to a $theme-colors variable we are making them available to use in utility classes for border/text/background colors

File: src/assets/images/arrow-down.svg (hunk 8 above)
Comment (ilyawzrd) [reply src/assets/images/arrow-down.svg:1]: done, removed unused arrow icons

File: src/app/modules/addresses/addresses.component.ts (hunk 7 above)
Comment (ilyawzrd) [reply src/app/modules/addresses/addresses.component.ts]: done

File: src/app/modules/addresses/addresses.component.html (hunk 3 above)
Comment (ilyawzrd) [reply src/app/modules/addresses/addresses.component.html:3]: done

File: src/app/graphql/queries/get-organization-addresses.graphql (hunk 2 above)
Comment (ilyawzrd) [reply src/app/graphql/queries/get-organization-addresses.graphql]: done

File: src/app/modules/addresses/addresses.component.ts (hunk 6 above)
Comment (ilyawzrd) [reply src/app/modules/addresses/addresses.component.ts]: done

File: src/app/modules/addresses/addresses.component.ts (hunk 4 above)
Comment (ilyawzrd) [reply src/app/modules/addresses/addresses.component.ts]: done

File: src/app/modules/addresses/addresses.component.ts (hunk 5 above)
Comment (ilyawzrd) [reply src/app/modules/addresses/addresses.component.ts]: moved to constants

**Review by asvishnyakov — APPROVED** [review]

**Pull request merged**